osmaxx/excerptexport/views.py

```diff
@@ -69,31 +73,91 @@ def get_object(self, queryset=None):
         return o
 
 
-class ExportsListView(LoginRequiredMixin, FrontendAccessRequiredMixin, ListView):
+class ExportsListMixin:
+    @property
+    def excerpt_ids(self):
+        if not hasattr(self, '_excerpt_ids'):
+            self._excerpt_ids = list(
+                OrderedSet(
+                    self.get_user_exports()
+                    .values_list('extraction_order__excerpt', flat=True)
+                )
+            )
+        return self._excerpt_ids
+
+    @property
+    def status_choices(self):
+        allowed_choices = [
+            statuses.FINISHED,
+            statuses.FAILED,
+        ]
+        return [choice for choice in Export.STATUS_CHOICES if choice[0] in allowed_choices]
+
+    def get_user_exports(self):
+        return self._filter_exports(
+            Export.objects.filter(extraction_order__orderer=self.request.user)
+        ).order_by('-updated', '-finished')
+
+    def _filter_exports(self, query):
+        allowed_status_filters = [
+            status[0] for status in self.status_choices
+        ]
+        status_filter = self.request.GET.get('status', None)
+        if status_filter and status_filter in allowed_status_filters:
```

Comment:
```python
if status_filter in allowed_status_filters:
```
should suffice here. `None` can be used on the left side of containment tests just fine. (`None in [None, 2, 'three']` is `True` while `None in [1, 2, 'three']` is `False`.)